tests/test.spec.ts

```diff
@@ -1,18 +1,78 @@
 import { join } from 'path';
 import * as rimraf from 'rimraf';
 import * as shell from 'shelljs';
+import { newImportStatementRegex, newStringRegex } from '../dist/utils';
 
 const projectsRoot = join(__dirname, '../projects');
 
-[1, 3, 4, 5, 6, 7, 8].forEach((value) => {
-  it(`Project ${value}`, () => {
-    const projectDir = join(projectsRoot, `project${value}`);
-    rimraf.sync(join(projectDir, 'dist'));
-    const { code, stderr } = shell.exec('npm start', {
-      cwd: projectDir,
-      silent: true
-    });
-    if (code !== 0) console.error(stderr);
-    expect(code).toEqual(0);
+/**
+ * Sample JavaScript code with a bunch of require/import
+ * statements with various spacing etc. Import paths from
+ * valid statements are incrementing numbers, starting from 0,
+ * so that it is easy to verify validity of results.
+ */
+const sampleImportStatements = `
+const module = require('0')
+var module = require
+(
+  '1'
+)  ;
+import module from ('2');
+import "3"
+
+imported ("invalid/import")
+
+import theDefault, {namedExport} from ("4")
+import {
+  extraLinesOhNo
+} from '5'
+const asyncImport = await import('6');
+
+export * from '7';
+
+import
+
+  '8'
+
+const notAnImport = unimport('something');
+`;
+
+function runTestProject(projectNumber: number) {
+  const projectDir = join(projectsRoot, `project${projectNumber}`);
```

Comment:
I originally pulled this out into its own function because I misunderstood how the test projects worked, and wanted to be able to run this in different contexts.